platforms/core-execution/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java

```diff
@@ -72,17 +72,20 @@ public class DefaultWorkerExecutor implements WorkerExecutor {
     private final Instantiator instantiator;
     private final IsolationScheme<WorkAction<?>, WorkParameters> isolationScheme = new IsolationScheme<>(Cast.uncheckedCast(WorkAction.class), WorkParameters.class, WorkParameters.None.class);
     private final CachedClasspathTransformer classpathTransformer;
-    private final File baseDir;
+    private final File projectDir;
     private final ProjectCacheDir projectCacheDir;
 
+    private final File settingsDir;
+
     public DefaultWorkerExecutor(
         WorkerFactory daemonWorkerFactory, WorkerFactory isolatedClassloaderWorkerFactory, WorkerFactory noIsolationWorkerFactory,
         JavaForkOptionsFactory forkOptionsFactory, WorkerThreadRegistry workerThreadRegistry, BuildOperationRunner buildOperationRunner,
         AsyncWorkTracker asyncWorkTracker, WorkerDirectoryProvider workerDirectoryProvider, WorkerExecutionQueueFactory workerExecutionQueueFactory,
         ClassLoaderStructureProvider classLoaderStructureProvider, ActionExecutionSpecFactory actionExecutionSpecFactory, Instantiator instantiator,
         CachedClasspathTransformer classpathTransformer,
-        File baseDir,
-        ProjectCacheDir projectCacheDir
+        File projectDir,
+        ProjectCacheDir projectCacheDir,
+        File settingsDir
```

Comment:
💅 I think it would be better if the order of these parameters would be `settingsDir, baseDir, projectCacheDir` instead. Also, it would make things easier to read if all involved types used the same order.